src/main/java/io/gravitee/policy/oauth2/Oauth2Policy.java

```diff
@@ -99,10 +97,23 @@ public boolean requireSubscription() {
     }
 
     @Override
-    public Completable onInvalidSubscription(HttpExecutionContext ctx) {
-        return ctx.interruptWith(
-            new ExecutionFailure(UNAUTHORIZED_401).key(GATEWAY_OAUTH2_ACCESS_DENIED_KEY).message(OAUTH2_ERROR_ACCESS_DENIED)
-        );
+    public Maybe<SecurityToken> extractSecurityToken(HttpExecutionContext ctx) {
+        final OAuth2Resource<?> oauth2Resource = getOauth2Resource(ctx);
+        if (oauth2Resource == null) {
+            log.debug("Skipping security token extraction cause no oauth2 resource configured");
+            return Maybe.empty();
+        }
```

Comment:
Does it makes sens to move this into `extractAccessToken` so we can benefit of it when using the policy outside a security policy (ie: in a Flow) ? WDYT ?

Reply:
When there is no oauth2 resource configured, the behavior is different :
- during plan selection, extractSecurityToken, returns empty without raising an error
- during policy execution, handleSecurity interrupts with a 401 with a specific error message

So I think we shouldn't mutualize in `extractAccessToken` if we want to keep this distinction.
And also, wouldn't really makes sense to me as extracting the access token from incoming request doesn't need any oauth2 resource at all.